config/module.go

```diff
@@ -14,6 +14,10 @@ var moduleNameRegEx = regexp.MustCompile(`^[\w-]+$`)
 const reservedModuleName = "parent"
 
 // Module represents an external resource module, with a path to the binary module file.
+//
+// Note to fellow programmers:
+// If you add any new fields, make sure to bring them to the modmanager.module struct and the modmanager.Configs() function
+// Otherwise there will be config diffs on every reconfigure.
```

Comment:
Ah yeah sorry for not catching this in the earlier PR (I don't remember which one, but I think I at least glanced at the original changes). Thanks for this note. 

The only other method we could use to avoid this issue is storing a `moduleConfigs []config.Module` field on the module manager for all added modules and returning that slice for `Configs` instead of a constructed one from `mgr.modules`. That strategy has the disadvantage of requiring more state management for the module manager (you'd have to modify `moduleConfigs` in `add` and `remove`). I'm assuming this is the end of new fields on `config.Module` for a while, so I don't have a super strong opinion.

Reply:
Oh I see you suggested that method in the PR description. Do you want to try storing `moduleConfigs []config.Module` on the manager instead? A unit test for `Configs` would be helpful, too.

Reply:
why can't a modmanager.module store a config.Module as a attribute?

like 
```
type modmanager.module struct {
 cfg config.Module
}
```

Reply:
This seems like the simplest approach

Reply:
Ah there's no reason to not do that; that should work.

Reply:
Okay -- cool. I'm planning on merging this once the tests pass and we can look into refactoring this later. (I'm going on vacation tomorrow and would like to get this in before then if possible)